File: EulerInAny/Euler1.ml
open Core.Std;;

let ofThreeOrFive x = x % 3 = 0 || x % 5 = 0;;

let range x y = 
	let rec ranger x y acc =
		match y with
			|y when x > y  -> []
			|y when x = y -> (y::acc)
			|y -> ranger x (y - 1) (y::acc)
	in ranger x y [];;

let divisible x = List.filter ~f:ofThreeOrFive (range 1 999);;

let answer = List.fold ~f:(+) ~init:0 (divisible ());;

printf "%d\n" answer;;

File: rust/Euler1.rs
fn is_mod3(x: int) -> bool {
	if x % 3i == 0i { true } else { false };
}

fn is_mod5(y: int) -> bool {
	if y % 5i == 0i { true } else { false };
}

fn main() {
	let mut answer = 0;
	for i in range(0i, 1000i) {
        if is_mod3(i) || is_mod5(i) {
        	answer += i;
        }
	}
	print!("{}", answer);
}